core/server/api/themes.js

```diff
@@ -68,6 +68,16 @@ themes = {
                 // We use the model, not the API here, as we don't want to trigger permissions
                 return settingsModel.edit(newSettings, options);
             })
+            // Load theme translations when activating a theme
+            // (when starting Ghost, or when switching themes).
+            // There is usually no need to reload the back-end
+            // translations (if enabled) with i18n.init()
+            // when switching themes, because the old theme
+            // translations take little memory space and do
+            // not conflict (they have different theme name).
+            .then(function () {
+                return i18n.loadThemeTranslations();
```

Comment:
Same here, we can listen on the event if the active theme get's changed.
Example code which you can use in the i18n module: https://github.com/TryGhost/Ghost/blob/master/core/server/models/base/listeners.js#L51

Reply:
Possibly that's not needed, since that's in the `activate()` function of _themes.js_, only called when activating a theme, if I'm not wrong.